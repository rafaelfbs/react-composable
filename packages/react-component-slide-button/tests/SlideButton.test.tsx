import * as React from "react";
import * as renderer from 'react-test-renderer';
import SlideButton from "../src";

describe('<SlideButton />', () => {
    describe('with default props', () => {
        it('renders correctly', () => {
            const tree = renderer.create(<SlideButton/>).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('with different orientation', () => {
        it('accepts "left" orientation', () => {
            const tree = renderer.create(
                <SlideButton
                    theme={{...SlideButton.defaultProps.theme, orientation: 'left'}}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('accepts "right" orientation', () => {
            const tree = renderer.create(
                <SlideButton
                    theme={{...SlideButton.defaultProps.theme, orientation: 'right'}}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('with custom colors', () => {
        it('accepts "primary" color', () => {
            const tree = renderer.create(
                <SlideButton
                    theme={{...SlideButton.defaultProps.theme, primary: 'red'}}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('accepts "secondary" color', () => {
            const tree = renderer.create(
                <SlideButton
                    theme={{...SlideButton.defaultProps.theme, secondary: 'red'}}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('with custom border width', () => {
        it('accepts custom "borderWidth"', () => {
            const tree = renderer.create(
                <SlideButton
                    theme={{...SlideButton.defaultProps.theme, borderWidth: '1rem'}}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('accepts zero "borderWidth"', () => {
            const tree = renderer.create(
                <SlideButton
                    theme={{...SlideButton.defaultProps.theme, borderWidth: '0'}}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('with custom transition duration', () => {
        it('accepts custom "transitionDuration"', () => {
            const tree = renderer.create(
                <SlideButton
                    theme={{...SlideButton.defaultProps.theme, transitionDuration: '0.5s'}}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        })
    })
});
