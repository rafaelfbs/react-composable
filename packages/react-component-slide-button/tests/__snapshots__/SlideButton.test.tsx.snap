// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideButton /> with custom border width accepts custom "borderWidth" 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  text-align: inherit;
  background-size: 200% 100%;
  background-image: linear-gradient( to right, #F0F0F0 50%, #026DB2 50% );
  -webkit-transition: background-position 0.1s linear;
  transition: background-position 0.1s linear;
  border: none;
  border-left: 1rem solid #026DB2;
}

.c0:hover {
  background-position: -100% 0;
}

<button
  className="c0"
/>
`;

exports[`<SlideButton /> with custom border width accepts zero "borderWidth" 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  text-align: inherit;
  background-size: 200% 100%;
  background-image: linear-gradient( to right, #F0F0F0 50%, #026DB2 50% );
  -webkit-transition: background-position 0.1s linear;
  transition: background-position 0.1s linear;
  border: none;
  border-left: 0 solid #026DB2;
}

.c0:hover {
  background-position: -100% 0;
}

<button
  className="c0"
/>
`;

exports[`<SlideButton /> with custom colors accepts "primary" color 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  text-align: inherit;
  background-size: 200% 100%;
  background-image: linear-gradient( to right, red 50%, #026DB2 50% );
  -webkit-transition: background-position 0.1s linear;
  transition: background-position 0.1s linear;
  border: none;
  border-left: 5px solid #026DB2;
}

.c0:hover {
  background-position: -100% 0;
}

<button
  className="c0"
/>
`;

exports[`<SlideButton /> with custom colors accepts "secondary" color 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  text-align: inherit;
  background-size: 200% 100%;
  background-image: linear-gradient( to right, #F0F0F0 50%, red 50% );
  -webkit-transition: background-position 0.1s linear;
  transition: background-position 0.1s linear;
  border: none;
  border-left: 5px solid red;
}

.c0:hover {
  background-position: -100% 0;
}

<button
  className="c0"
/>
`;

exports[`<SlideButton /> with custom transition duration accepts custom "transitionDuration" 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  text-align: inherit;
  background-size: 200% 100%;
  background-image: linear-gradient( to right, #F0F0F0 50%, #026DB2 50% );
  -webkit-transition: background-position 0.5s linear;
  transition: background-position 0.5s linear;
  border: none;
  border-left: 5px solid #026DB2;
}

.c0:hover {
  background-position: -100% 0;
}

<button
  className="c0"
/>
`;

exports[`<SlideButton /> with default props renders correctly 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  text-align: inherit;
  background-size: 200% 100%;
  background-image: linear-gradient( to right, #F0F0F0 50%, #026DB2 50% );
  -webkit-transition: background-position 0.1s linear;
  transition: background-position 0.1s linear;
  border: none;
  border-left: 5px solid #026DB2;
}

.c0:hover {
  background-position: -100% 0;
}

<button
  className="c0"
/>
`;

exports[`<SlideButton /> with different orientation accepts "left" orientation 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  text-align: inherit;
  background-size: 200% 100%;
  background-image: linear-gradient( to right, #F0F0F0 50%, #026DB2 50% );
  -webkit-transition: background-position 0.1s linear;
  transition: background-position 0.1s linear;
  border: none;
  border-left: 5px solid #026DB2;
}

.c0:hover {
  background-position: -100% 0;
}

<button
  className="c0"
/>
`;

exports[`<SlideButton /> with different orientation accepts "right" orientation 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  color: inherit;
  font: inherit;
  line-height: normal;
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;
  -webkit-appearance: none;
  text-align: inherit;
  background-size: 200% 100%;
  background-image: linear-gradient( to left, #026DB2 50%, #F0F0F0 50% );
  -webkit-transition: background-position 0.1s linear;
  transition: background-position 0.1s linear;
  border: none;
  border-right: 5px solid #026DB2;
}

.c0:hover {
  background-position: 100% 0;
}

<button
  className="c0"
/>
`;
